### ISSUE TITLE
[CI] ChangePolicyforIndexIT#testChangePolicyForIndex fails

### ISSUE THREAD

**Issue by jimczi** [elastic/elasticsearch#35244]
Two failures in 6x, however this does not reproduce for me:

```
org.junit.ComparisonFailure: expected:<[completed]> but was:<[hot]>
	at __randomizedtesting.SeedInfo.seed([E85178511074E054:9402C10DD3133A4]:0)
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.elasticsearch.xpack.indexlifecycle.ChangePolicyforIndexIT.assertStep(ChangePolicyforIndexIT.java:140)
	at org.elasticsearch.xpack.indexlifecycle.ChangePolicyforIndexIT.lambda$testChangePolicyForIndex$2(ChangePolicyforIndexIT.java:115)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:848)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:822)
	at org.elasticsearch.xpack.indexlifecycle.ChangePolicyforIndexIT.testChangePolicyForIndex(ChangePolicyforIndexIT.java:115)
```

The build was https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+multijob-unix-compatibility/os=opensuse/39/console:

```
REPRODUCE WITH: ./gradlew :x-pack:plugin:ilm:qa:multi-node:integTestRunner \
  -Dtests.seed=E85178511074E054 \
  -Dtests.class=org.elasticsearch.xpack.indexlifecycle.ChangePolicyforIndexIT \
  -Dtests.method="testChangePolicyForIndex" \
  -Dtests.security.manager=true \
  -Dtests.locale=en-SG \
  -Dtests.timezone=America/Nome \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

Another run with the same issue:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+multijob-unix-compatibility/os=ubuntu&&virtual/36/console

**Comment by talevy** [comment]
thanks @jimczi. I'll take a look at this

**Comment by javanna** [comment]
Another failure, this one from master: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/149

**Comment by talevy** [comment]
thanks @javanna, this looks like another error up the change manifested by the change that meant to address the original issue.

I will mute the test and dig into this

relevant logs to debug:

```
[2018-11-06T01:05:38,088][ERROR][o.e.x.i.ExecuteStepsUpdateTask] [node-0] policy [policy_2] for index [test-000001] failed on cluster state step [{"phase":"hot","action":"rollover","name":"update-rollover-lifecycle-date"}]. Moving to ERROR step
```

**Comment by talevy** [comment]
test muted in master: https://github.com/elastic/elasticsearch/commit/2bf843e768f65d2531ad97fcb8c0b91779182913, and 6.x: https://github.com/elastic/elasticsearch/commit/c94d02a1b79ad02fb3b95a7a37b863eda49d31cb

**Comment by gwbrown** [comment]
The instability is fixed and the test reenabled by #35524
